Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -309,60 +318,40 @@ void FieldProbe::ComputeDiags (int step)
+                        part_Bz[ip] = Bzp; //remember to add lorentz transform
                         part_S[ip] = S; //remember to add lorentz transform
                     }
                 });// ParallelFor Close
                 if (m_field_probe_integrate)
                 {
-                    amrex::Real const dt = WarpX::GetInstance().getdt(lev);
-                    if (m_intervals.contains(step+1))
-                    {
-                        amrex::Real time_ellapsed {m_iterated_steps * dt};
-                        for (int ip=0; ip < np; ip++)
-                        {
-                            // Fill output array
-                            // TODO replace with explicit copies later on to avoid managed memory on GPU
-                            m_data[0 * noutputs + ParticleVal::Ex] = part_Ex[ip] * time_ellapsed;
-                            m_data[0 * noutputs + ParticleVal::Ey] = part_Ey[ip] * time_ellapsed;
-                            m_data[0 * noutputs + ParticleVal::Ez] = part_Ez[ip] * time_ellapsed;
-                            m_data[0 * noutputs + ParticleVal::Bx] = part_Bx[ip] * time_ellapsed;
-                            m_data[0 * noutputs + ParticleVal::By] = part_By[ip] * time_ellapsed;
-                            m_data[0 * noutputs + ParticleVal::Bz] = part_Bz[ip] * time_ellapsed;
-                            m_data[0 * noutputs + ParticleVal::S] = part_S[ip] * time_ellapsed;
-                        }
-                    }
-                    m_iterated_steps++;
+                    if (!m_intervals.contains(step+1)) { return; }
                 }
-                else
+                for (int ip=0; ip < np; ip++)
                 {
-                    for (int ip=0; ip < np; ip++)
-                    {
-                        // Fill output array
-                        m_data[0 * noutputs + ParticleVal::Ex] = part_Ex[ip];
-                        m_data[0 * noutputs + ParticleVal::Ey] = part_Ey[ip];
-                        m_data[0 * noutputs + ParticleVal::Ez] = part_Ez[ip];
-                        m_data[0 * noutputs + ParticleVal::Bx] = part_Bx[ip];
-                        m_data[0 * noutputs + ParticleVal::By] = part_By[ip];
-                        m_data[0 * noutputs + ParticleVal::Bz] = part_Bz[ip];
-                        m_data[0 * noutputs + ParticleVal::S] = part_S[ip];
-                    }
+                    // Fill output array
+                    m_data[0 * noutputs + FieldProbePIdx::Ex] = part_Ex[ip];
+                    m_data[0 * noutputs + FieldProbePIdx::Ey] = part_Ey[ip];
+                    m_data[0 * noutputs + FieldProbePIdx::Ez] = part_Ez[ip];
+                    m_data[0 * noutputs + FieldProbePIdx::Bx] = part_Bx[ip];
+                    m_data[0 * noutputs + FieldProbePIdx::By] = part_By[ip];
+                    m_data[0 * noutputs + FieldProbePIdx::Bz] = part_Bz[ip];
+                    m_data[0 * noutputs + FieldProbePIdx::S] = part_S[ip];
```

Comment:
Already preparing for np>1 particles:
```suggestion
                    m_data[ip * noutputs + FieldProbePIdx::Ex] = part_Ex[ip];
                    m_data[ip * noutputs + FieldProbePIdx::Ey] = part_Ey[ip];
                    m_data[ip * noutputs + FieldProbePIdx::Ez] = part_Ez[ip];
                    m_data[ip * noutputs + FieldProbePIdx::Bx] = part_Bx[ip];
                    m_data[ip * noutputs + FieldProbePIdx::By] = part_By[ip];
                    m_data[ip * noutputs + FieldProbePIdx::Bz] = part_Bz[ip];
                    m_data[ip * noutputs + FieldProbePIdx::S] = part_S[ip];
```